### ISSUE TITLE
OptionsBuilder EFcore UseSqlite Error: Keyword not supported

### ISSUE THREAD

**Issue by BryantAvey** [ericsink/SQLitePCL.raw#147]
Eric, I'm get this error:
System.ArgumentException: Keyword not supported: 'password'.

While trying to do something like this:
using Microsoft.EntityFrameworkCore;
using PCLStorage;
using SQLitePCL;
        
namespace com.Company.Product.Mobile.Client.Services
{
    public class SQLiteContext : DbContext
    {
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            const string password = "MyPassword";
            const string filename = "SQLCipherMobileClientDb.SQLite";
            var path = FileSystem.Current.LocalStorage.Path;
            raw.SetProvider(new SQLite3Provider_sqlcipher());
            optionsBuilder.UseSqlite($"Filename= {path}/{filename}; Password= {password}");
            base.OnConfiguring(optionsBuilder);
        }
    }
}

Basically I'm trying to just use SQLCipher and create an encrypted SQLite db in a Xamarin Droid project.

Is there a way or method or combo of SQLitePCL.raw libraries or bundles that would allow me to something like this?  It doesn't seem to be able to take the password parm.

I'd be grateful for any thoughts or advice on how to make an encrypted SQLite db using SQLCipher inside a Xamarin Forms .NETcore project.

I know Brice Lambson and you have been working on this for .NETcore V2, but I'm hoping you've maybe got an interim method that might work for us.

Thanks for all your work with this!

**Comment by ericsink** [comment]
To use this with Entity Framework, you need EF Core 2.0.  There isn't any interim method.

If you don't care about whether you are using EF or not and just want something that will do sqlcipher...

Wait, in one paragraph you say you want Xamarin Droid, in another you want .NET core.  Shouldn't matter *too* much actually.  But anyway...

The best high-level approach is:

1.  Start with a sample or app that just uses SQLite.

2.  Switch the library from SQLite to SQLCipher, with no changes to your code.

3.  Add code to use pragma key, to encrypt the db

For step 1, consider trying sqlite-net-pcl (that's a NuGet package id) instead of Entity Framework.  There should be lots of samples and apps out there that use it.

For step 2, https://github.com/ericsink/SQLitePCL.raw/wiki/How-to-use-SQLCipher-with-SQLite-net

For step 3, https://www.zetetic.net/sqlcipher/sqlcipher-api/

**Comment by BryantAvey** [comment]
Thanks Eric. I wanted to check.  We can't use the pragma key in Xamarin Forms so that option is out. Brice suggested that also, but Xamarin Forms doesn't support pragma. Unless you know of a way to do that somehow in a Xam Forms environment.
I had a version working fine with simple SQLCipher and Xamarin forms but we've converted everything to EF Core. So I think we'll have to just wait until Brice finishes EF Core 2.0.  Thanks man.  I'm hearing that crash and burn sizzling sound right now :)

**Comment by ericsink** [comment]
I see no reason you couldn't use pragma key in Xamarin Forms.

It's just a SQLite statement, and using Xamarin Forms (or not) should have nothing to do with whether you can execute that SQLite statement.

What *does* matter is what API or wrapper you are using to access SQLite.  That API should provide you a way to execute a SQLite statement.  And you should be able to use that to do a pragma key.

**Comment by BryantAvey** [comment]
Interesting. I thought I saw someone saying that it wasn't supported in a Xam Forms Droid target. But let me try and get back with my results.

**Comment by ericsink** [comment]
Reopen this or open a new issue if there is something here I need to address.
